contracts/protocore/OriginProtocore.sol

```diff
@@ -51,53 +51,39 @@ contract OriginProtocore is MasterCopyNonUpgradable, GenesisOriginProtocore, Pro
      * @dev These input params will be provided by the coconsensus contract.
      *      This can be called only by the coconsensus contract once.
      *
-     * @param _metachainId Metachain id.
-     * @param _domainSeparator Domain separator.
-     * @param _epochLength Epoch length.
-     * @param _metablockHeight Metablock height.
-     * @param _selfProtocore SelfProtocore contract address.
-     *
-     * \pre `_metachainId` is not 0.
-     * \pre `_domainSeparator` is not 0.
-     * \pre `_epochLength` is not 0.
-     * \pre `_selfProtocore` is not 0.
+     * @return Block hash and block number of finalized genesis checkpoint.
      *
      * \post Sets `selfProtocore` to the given value.
      * \post Sets `domainSeparator` to the given value.
      * \post Sets `epochLength` to the given value.
      * \post Sets `metachainId` to the given value.
      * \post Sets genesis link.
      */
-    function setup(
-        bytes32 _metachainId,
-        bytes32 _domainSeparator,
-        uint256 _epochLength,
-        uint256 _metablockHeight,
-        address _selfProtocore
-    )
+    function setup()
         external
         onlyCoconsensus
+        returns (bytes32, uint256)
```

Comment:
This is breaking coding style. For this scenario we should use named return params.

Reply:
Updated.